src/client/Client.cc

```diff
@@ -4142,6 +4153,17 @@ void Client::remove_session_caps(MetaSession *s)
       in->mark_caps_clean();
       put_inode(in.get());
     }
+    caps &= CEPH_CAP_FILE_CACHE | CEPH_CAP_FILE_BUFFER;
+    if (err == -EBLACKLISTED &&
+	cct->_conf.get_val<bool>("client_discard_dirty_on_blacklisted") &&
+	caps && !in->caps_issued_mask(caps, true)) {
+      if (in->oset.dirty_or_tx) {
+	lderr(cct) << __func__ << " still has dirty data on " << *in << dendl;
+	in->set_async_err(err);
+      }
+      objectcacher->purge_set(&in->oset);
```

Comment:
What happens when a future `fsync(fd)` comes? I may be missing it, but we would want `-EIO` in that case. Yes?

Reply:
the set_async_err() above make future fsync/close return error.